static/js/compose.js

```diff
@@ -560,7 +573,13 @@ function send_message(request) {
         }
     }
     var locally_echoed = local_id !== undefined;
-
+    exports.msg_server_status[local_id] = "failed";
+    setTimeout(function () {
```

Comment:
Do we want to have a mechanism to clear this timeout if the message get successfully delivered?  It may be ultimately cleaner that way.